core/controller/src/main/scala/whisk/core/loadBalancer/DistributedLoadBalancerData.scala

```diff
@@ -17,69 +17,102 @@
 
 package whisk.core.loadBalancer
 
+import akka.actor.Actor
+import akka.actor.ActorRef
 import akka.actor.ActorSystem
+import akka.actor.Props
 import akka.util.Timeout
-import akka.pattern.ask
-import whisk.common.Logging
-import whisk.core.entity.{ActivationId, UUID}
-
-import scala.collection.concurrent.TrieMap
-import scala.concurrent.Future
+import scala.collection.mutable
 import scala.concurrent.duration._
+import whisk.common.Logging
+import whisk.core.entity.ActivationId
+import whisk.core.entity.InstanceId
+import whisk.core.entity.UUID
 
 /**
  * Encapsulates data used for loadbalancer and active-ack bookkeeping.
  *
  * Note: The state keeping is backed by distributed akka actors. All CRUDs operations are done on local values, thus
  * a stale value might be read.
  */
-class DistributedLoadBalancerData(implicit actorSystem: ActorSystem, logging: Logging) extends LoadBalancerData {
+class DistributedLoadBalancerData(instance: InstanceId, monitor: Option[ActorRef] = None)(
```

Comment:
What's `monitor` used for?

Reply:
monitor is used for testing, to test replication Changed events from SharedDataService:
```    case c @ Changed(e) =>
      logging.debug(this, "Current elements: " + c.get(storage))
      val res = c.get(storage).entries.mapValues(_.entries.mapValues(_.toInt))
      if (res.nonEmpty) {
        res.values.foreach(_.values.foreach(i => {
          require(i >= 0, s"values cannot be less than 0 ${res}")
        }))
        monitor ! Updated(storageName, res)
      }
```